collector/benthos/output/openmeter.go

```diff
@@ -159,14 +207,17 @@ func (o *openmeterOutput) WriteBatch(ctx context.Context, batch service.MessageB
 		return err
 	}
 
+	span.SetAttributes(attribute.Int("openmeter.http.status_code", resp.StatusCode()))
+
 	// TODO: improve error handling
 	if resp.StatusCode() != http.StatusNoContent {
 		if err = resp.ApplicationproblemJSON400; err != nil {
 			return err
 		} else if err = resp.ApplicationproblemJSONDefault; err != nil {
 			return err
 		} else {
-			return fmt.Errorf("unknown error: %w", err)
+			err = fmt.Errorf("unknown error: %w", err)
+			return err
```

Comment:
Merge this into a single line?

Reply:
It is done on purpose so we can access the wrapped error here: https://github.com/openmeterio/openmeter/pull/1036/files#diff-bcf2cc4283810524c2c9ba39471a535d8821d1f887552d602a1280bae0f0fd66R155

Reply:
Reworked this part to make it a bit more readable and also left a comment in the code explaining why error handling is done the way it is.